tests/test_meshtastic_utils_metadata.py

```
@@ -0,0 +1,902 @@
+            # Verify exception handling
+            self.assertFalse(result["success"])
+            self.assertEqual(result["firmware_version"], "unknown")
+            mock_logger.debug.assert_called_once()
+
+    def test_get_device_metadata_raise_on_error_reraises(self):
+        """raise_on_error=True should propagate getMetadata probe failures."""
+        mock_client = MagicMock()
+        mock_client.localNode.getMetadata.side_effect = Exception("Device error")
+        mock_client.localNode.iface.metadata = None
+
+        with self.assertRaisesRegex(Exception, "Device error"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)